src/popup/pages/Retip.vue

```diff
@@ -33,149 +36,196 @@
+  defineComponent,
+  onMounted,
+  ref,
+  watch,
+  computed,
+  Ref,
+} from '@vue/composition-api';
 import { SCHEMA } from '@aeternity/aepp-sdk';
 import { MAGNITUDE, AETERNITY_CONTRACT_ID } from '../utils/constants';
-import { convertToken, watchUntilTruthy } from '../utils';
-import deeplinkApi from '../../mixins/deeplinkApi';
-import maxAmountMixin from '../../mixins/maxAmountMixin';
+import { convertToken, watchUntilTruthy, rxJsObservableToVueState } from '../utils';
+import { useMaxAmount } from '../../composables/maxAmount';
+import { useDeepLinkApi } from '../../composables/deepLinkApi';
 import InputAmount from '../components/InputAmountV2.vue';
 import UrlStatus from '../components/UrlStatus.vue';
 import BtnMain from '../components/buttons/BtnMain.vue';
 import BalanceInfo from '../components/BalanceInfo.vue';
 
-export default {
+export default defineComponent({
+  name: 'Retip',
   components: {
     InputAmount, UrlStatus, BtnMain, BalanceInfo,
   },
-  mixins: [deeplinkApi, maxAmountMixin],
-  data: () => ({
-    tip: {},
-    formModel: {
+  setup(props, { root }) {
+    const { openCallbackOrGoHome } = useDeepLinkApi({ router: root.$router });
+
+    const tip = ref<any>({
+      url: 'default',
+    });
+    const formModel = ref<any>({
       amount: null,
       selectedAsset: null,
-    },
-    loading: false,
-  }),
-  computed: {
-    ...mapGetters(['account', 'tippingSupported']),
-    ...mapGetters('fungibleTokens', ['getAeternityToken']),
-    ...mapState({
-      tippingV1: 'tippingV1',
-      tippingV2: 'tippingV2',
-      urlStatus(state, getters) {
-        return getters['tipUrl/status'](this.tip.url);
-      },
-
-      validationStatus({ sdk }, { minTipAmount }) {
-        if (!sdk || !this.tippingContract) {
-          return { error: true };
-        }
-        if (this.formModel.selectedAsset.contractId !== AETERNITY_CONTRACT_ID
-          && this.$route.query.id.includes('_v1')) {
-          return { error: true, msg: this.$t('pages.tipPage.v1FungibleTokenTipError') };
-        }
-        if (this.formModel.selectedAsset.contractId === AETERNITY_CONTRACT_ID
-          && +this.formModel.amount < minTipAmount) {
-          return { error: true, msg: this.$t('pages.tipPage.minAmountError') };
-        }
-        return { error: false };
-      },
-    }),
-    tippingContract() {
-      return this.$route.query.id.includes('_v2') || this.$route.query.id.includes('_v3')
-        ? this.tippingV2
-        : this.tippingV1;
-    },
-  },
-  async created() {
-    this.loading = true;
-    this.formModel.selectedAsset = this.getAeternityToken({
-      tokenBalance: this.balance,
-      balanceCurrency: this.balanceCurrency,
     });
+    const {
+      max,
+      fee,
+      balance,
+      account,
+    } = useMaxAmount({
+      formModel,
+      store: root.$store,
+    });
+    const balanceCurrency = rxJsObservableToVueState(
+      root.$store.state.observables.balanceCurrency,
+    ) as Ref<any>;
```

Comment:
`balanceCurrency` in most places is a `number` so I guess having `any` here is a mistake.